starter_code/elevator.js

```diff
@@ -1,19 +1,68 @@
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = [];
+    this.waitingList = [];
+    this.passengers = [];
+    this.direction = 'up';
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start() {
+    var interval = setInterval(() => {
+      this.update();
+    }, 1000);
+  }
+  stop() {
+    clearInterval(interval);
+  }
+  update() {
+    if (this.direction === 'up')
+      this.floorUp();
+    else if (this.direction === 'down')
+      this.floorDown();
+    this.log();
+    _passengersEnter();
+    _passengersLeave();
+  }
+  _passengersEnter() {
+    let passengersIndex = [];
+    for (let i = 0; i < this.waitingList.length; i++)
+      if (this.waitingList[i].originFloor === this.floor) {
+        passengersIndex.push(i);
+      }
+
+    for (let i = passengersIndex.length - 1; i >= 0; i--) {
+      let passenger = this.waitingList.splice(passengersIndex[i], 1);
+      this.passengers.push(passenger);
+      this.requests.push(passenger.destinationFloor);
```

Comment:
Actually, `this.reqeusts` should hold all the origin floors that have been called, so it should be updated in the `call` method. Once the passenger enters, you actually need to remove it from the requests.